http.go

```diff
@@ -322,7 +329,87 @@ func httpPutViolation(w http.ResponseWriter, r *http.Request) {
 	// Force object field and type to match value specified in request path
 	v.Object = valstr
 	v.Type = typestr
-	httpPutViolationsInner(w, r, typestr, []ViolationRequest{v})
+	if err := putViolation(typestr, v); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	return
+}
+
+func putViolation(typestr string, v ViolationRequest) error {
```

Comment:
Why not just accumulate errors in `httpPutViolationsInner` and return them at the end of the function with the desired status code? It might be less code for the same result